apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -0,0 +1,86 @@
+import { TRPCError } from "@trpc/server"
+import { z } from "zod"
+
+import { protectedProcedure, router } from "~/server/trpc"
+import { db } from "../database"
+
+// NOTE: We want to accept string
+// but validate that the string conforms to bigint.
+// Oddly enough, kysely doesn't allow `bigint` to query
+const bigIntSchema = z
+  // NOTE: A valid `bigint` is one that
+  // begins with a non-zero digit
+  // and has length > 1
+  .string()
+  .min(1)
+  .regex(/^[0-9]+$/)
+  .refine((v) => v.at(0) !== "0")
```

Comment:
we shud use `node-pg-types` - see [here](https://github.com/brianc/node-pg-types) so that we can use js `Number` seamlessly without the extra conversion(s)

Reply:
Cannot convert bigint to numbers. it will be lossy. we should just use uuids instead of bigints lol.

Reply:
i don't convert ah - we're still using bigint. this is just validating that the value is a numerical string that is greater than 0